### DISCUSSION TITLE
Quiz

### DISCUSSION THREAD

**Discussion by f1v3-dev** [NHN-Team-03/ITP#44] [General]
## 아래의 내용을 정리해보세요

1. **Definition** vs. **Declaration**
2. **Initialization** vs. **Assignment**
3. **Expression** vs. **Statement**

**Comment by P-C-Space** [comment]
2. Initialization vs Assignment ( 초기화 vs 할당 )
- Assignment(할당)은 " variable에 값을 넣어준다. " 라는 의미이다.
- Initialization(초기화)는 포커게임에서도 알 수 있다. 말 그대로 "초기값"으로 처음으로 할당한 값이다. 

1. Definition vs Declaration ( 정의 vs 선언 )
- Declaration(선언) 프로그래밍언어에서 Type과 variable-name을 "선언"한다는 의미이다.
- Definition(정의)는 Declaration(선언)과 Initialzaion(초기화)를 합친 개념으로 볼 수 있습니다.

3. Expression vs Statement ( 표현식 vs 문장 )
- Expression은 값을 반확하는 "식"으로 "x * 5"를 예로 들 수 있다.
- Statement는 프로그램에서 동작하는 문장으로 " x = x * 5; " 이는 Expression-Statement 라고 말할 수 있다.

**Comment by f1v3-dev** [comment]
## 1. Definition vs. Declaration
- Definition(정의): 메모리를 할당한다.
- Declaration(선언): 메모리를 할당하지 않고, 대상의 이름만 알려준다

> 둘의 가장 큰 차이는 '메모리를 할당하는가?' 이다.

## 2. Initialization vs. Assignment
- Initialization(초기화): 변수가 처음 정의되거나 선언될 때 변우세 초기값을 할당하는 과정
- Assignment(할당): 정의된 변수의 값을 변경하는 작업
 
 
## 3. Expression vs. Statement
- Expression(표현식): 특정한 결과 값, **변하지 않음!**
- Statement(문): 실행 가능한 최소의 독립적인 코드 조각, **상태가 변함!**

```java
int i; // Declaration
6 + 9 + 8; // Expression
i = 6 + 9 + 8; // Statement
```
